user_service/app/views.py

```diff
@@ -0,0 +1,217 @@
+            return web.json_response({"resp": "reg user fail"})
+
+        ph = argon.PasswordHasher()
+        password = ph.hash(body_dict["password"])
+
+        body_dict['uuid'] = str(uuid.uuid4())
+        body_dict['password'] = password
+
+        async with self.request.app["database_pool"].acquire() as conn:
+            await conn.execute(sa.insert(Users).values(**body_dict))
+
+        return web.json_response({"resp": f"add user {body_dict['username']}"})
+
+
+class LogInView(web.View):
+
+    async def post(self):
+        body_dict = await self.request.json()
+
+        async with self.request.app["database_pool"].acquire() as conn:
+            user = await conn.fetchrow(
+                sa.select([Users]).where(Users.username == body_dict['username']))
+
+            if user is not None:
+                pass_crypt = user["password"]
+                ph = argon.PasswordHasher()
+
+                try:
+                    ph.verify(pass_crypt, body_dict["password"])
+                except argon.exceptions.VerifyMismatchError:
+                    return web.json_response({"resp": "wrong password"})
+
+                a_token = jwt.encode(
+                    {
+                        "username": body_dict['username'],
+                        "exp": datetime.now() + timedelta(minutes=15),
+                        "uuid": user["uuid"]
+                    },
+                    settings.jwt_phrase)
+                r_token = jwt.encode(
+                    {
+                        "username": body_dict['username'],
+                        "exp": datetime.now() + timedelta(minutes=40),
+                        "uuid": user["uuid"]
+                    },
+                    settings.jwt_phrase)
+
+                redis = self.request.app["redis_pool"]
+                await redis.set(user["uuid"], " ".join([a_token, r_token]))
+                await redis.expire(user["uuid"], 60 * 40)
+
+                return web.json_response({"atoken": a_token, "rtoken": r_token})
+
+        return web.json_response({"resp": "login fail"})
+
+
+class LogOutView(web.View):
+    @auth
+    async def post(self, access_token_jwt):
+        redis = await self.request.app["redis_pool"]
+        await redis.delete(access_token_jwt["uuid"])
+        return web.json_response({"resp": "ok"})
+
+
+class UpdateView(web.View):
+    @auth
+    async def post(self, access_token_jwt):
+        body_dict = await self.request.json()
+        username = access_token_jwt["username"]
+        pool = self.request.app["database_pool"]
+
+        # gen password hash
+        if "password" in body_dict:
+            ph = argon.PasswordHasher()
+            password = ph.hash(body_dict["password"])
+            body_dict["password"] = password
+
+        # update user info
+        async with pool.acquire() as conn:
+            query = sa.update(Users).where(Users.username == username).values(**body_dict)
+            await conn.fetchrow(query)
```

Comment:
Если сработает такой вариант, то лучше так. Это не пересылает лишние байтики обратно по сети.
```suggestion
            await conn.execute(query)
```

Reply:
работает
https://github.com/ssnk-test/stage_proj/pull/10/commits/d45b360e97ba8c8fef0b0c0b0aa38090770c1551